core/src/main/java/com/opensymphony/xwork2/security/DefaultAcceptedPatternsChecker.java

```diff
@@ -35,7 +37,11 @@
     private static final Logger LOG = LogManager.getLogger(DefaultAcceptedPatternsChecker.class);
 
     public static final String[] ACCEPTED_PATTERNS = {
-            "\\w+((\\.\\w+)|(\\[\\d+\\])|(\\(\\d+\\))|(\\['(\\w|[\\u4e00-\\u9fa5])+'\\])|(\\('(\\w|[\\u4e00-\\u9fa5])+'\\)))*"
+            "\\w+((\\.\\w+)|(\\[\\d+])|(\\(\\d+\\))|(\\['(\\w|[\\u4e00-\\u9fa5])+'])|(\\('(\\w|[\\u4e00-\\u9fa5])+'\\)))*"
```

Comment:
Looks strange for me. Closing escaped [ brackets with not-escaped ] brackets?

Reply:
It was suggested by IDEA, restored the previous version :)

Reply:
Ah sorry [it was right](https://stackoverflow.com/questions/43044918/why-closing-square-bracket-doesnt-require-escaping-in-java-regex).

And I spotted it's nice I think to refactor `((\\.\\w+)|(\\[\\d+\\])|(\\(\\d+\\))|(\\['(\\w|[\\u4e00-\\u9fa5])+'\\])|(\\('(\\w|[\\u4e00-\\u9fa5])+'\\)))*` part e.g. named `EXTENSION_PATTERN` or so because it's same in both.

Reply:
Yes, but I would like to keep them as is, it's easier to understand what's going on. Also future changes to the patterns want affect each other.